File: Back/NodeJS/modules/users/src/infrastructure/dataAccess/UserEntity.ts
import { Column, Entity } from "typeorm";
import { UserSnapshot } from "../../domain/users/UserSnapshot";
import 'reflect-metadata';

@Entity({
    schema: 'usr',
    name: 'users'
})
export class UserEntity {
    static Create(userSnapshot: UserSnapshot): UserEntity {
        const entity = new UserEntity();
        entity.Email = userSnapshot.Email;
        entity.Id = userSnapshot.UserId;
        entity.Username = userSnapshot.Username;
        entity.Password = userSnapshot.Password;
        entity.RefreshToken = userSnapshot.RefreshToken;
        return entity;
    }
    @Column({
        name: 'id',
        type: 'uniqueidentifier',
        primary: true
    })
    Id: string;
    @Column({
        name: 'username',
        type: 'varchar'
    })
    Username: string;
    @Column({
        name: 'email',
        type: 'varchar'
    })
    Email: string;
    @Column({
        name: 'password',
        type: 'varchar'
    })
    Password: string;
    @Column({
        name: 'refresh_token',
        type: 'varchar'
    })
    RefreshToken: string;
}

File: Back/.Net/Modules/Users/BetFriends.Users.Infrastructure/Repositories/Sql/DataAccess/UserEntity.cs
using BetFriends.Users.Domain.Users;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BetFriends.Users.Infrastructure.Repositories.Sql.DataAccess;

[Table("users", Schema = "usr")]
public class UserEntity
{
    public UserEntity()
    {
        
    }

    public UserEntity(UserSnapshot user)
    {
        Id = user.Id;
        Username = user.Username;
        Email = user.Email;
        Password = user.Password;
        RefreshToken = user.RefreshToken;
    }

    [Column("id", TypeName = "uniqueidentifier"), Key, Required]
    public Guid Id { get; init; }
    [Column("username", TypeName = "varchar(MAX)")]
    public string Username { get; set; }
    [Column("email", TypeName = "varchar(MAX)")]
    public string Email { get; set; }
    [Column("password", TypeName = "varchar(MAX)")]
    public string Password { get; set; }
    [Column("refresh_token", TypeName = "varchar(MAX)")]
    public string RefreshToken { get; set; }
}
